### PR TITLE
Add wait_screen_change option on send_key

### PR DESCRIPTION
**Pull request by okurz**
While doing this, deprecating do_wait which is calling wait_idle which we want
to avoid as it is too much relying on a backend specific implementation which
can not be made available on enough backends.

### CODE REVIEW

**Comment by okurz** [comment]
checks fixed

File: testapi.pm [hunk 1]
```diff
@@ -1025,11 +1031,24 @@ Special characters naming:
...
-    query_isotovideo('backend_send_key', {key => $key});
-    wait_idle() if $do_wait;
+    my $key = shift;
+    my %args;
+    if (@_ == 1) {
+        %args = (do_wait => $_[0]);
+    }
+    else {
+        %args = @_;
+    }
```
Comment (mudler) [inline testapi.pm]: isn't this equivalent?
```perl
my %args = (@_ == 1) ? (do_wait => $_[0]) : @_;
```

Comment (mudler) [reply testapi.pm]: since i see you are not using @_ anymore, you could also:
```perl
my %args = (@_ == 1) ? (do_wait => shift) : @_;
```

Comment (okurz) [reply testapi.pm]: good idea, done. PR updated

File: testapi.pm [hunk 2]
```diff
@@ -1025,11 +1031,18 @@ Special characters naming:
...
+    $args{do_wait}            //= 0;
+    $args{wait_screen_change} //= 0;
+    bmwqemu::log_call(key => $key, %args);
+    if ($args{wait_screen_change}) {
+        wait_screen_change { query_isotovideo('backend_send_key', {key => $key}) };
+    }
+    else {
+        query_isotovideo('backend_send_key', {key => $key});
+    }
+    wait_idle() if $args{do_wait};
```
Comment (foursixnine) [inline testapi.pm]: If we want to slowly remove wait_idle, why not also cluck if do_wait is > 0?.

Comment (okurz) [reply testapi.pm]: because that is already within wait_idle, isn't it?

**Pull request closed without merging**